src/dotnet/Chat.Service/Chats.cs

```diff
@@ -412,7 +441,24 @@ private async Task<ChatEntry> GetChatEntry(Session session, string chatId, long
     }
 
     private async Task JoinChat(Session session, string chatId, CancellationToken cancellationToken)
-        => await ChatAuthorsBackend.GetOrCreate(session, chatId, cancellationToken).ConfigureAwait(false);
+    {
+        var chatAuthor = await ChatAuthorsBackend.GetOrCreate(session, chatId, cancellationToken).ConfigureAwait(false);
+        if (chatAuthor.HasLeft) {
+            var command = new IChatAuthorsBackend.ChangeHasLeftCommand(chatAuthor.Id, false);
+            await Commander.Call(command, cancellationToken).ConfigureAwait(false);
+        }
+        if (await IsInvited(session, chatId, cancellationToken).ConfigureAwait(false)) {
+            var updateOptionCommand1 = new ISessionOptionsBackend.UpsertCommand(
```

Comment:
why not pass multiple options into a single command? I see that https://github.com/Actual-Chat/actual-chat/blob/3381910bcc566e716dd51e8fb8f6b038f6242acc/src/dotnet/Users.Service/SessionOptionsBackend.cs#L28 can support multiple options - we just need to pass them into command instance